dockyard/common/utils.py

```diff
@@ -0,0 +1,68 @@
+from oslo_config import cfg
+
+from dockyard.common import base, link
+from dockyard.common.container.scheduler.round_robin import RoundRobinScheduler
+
+from urllib3 import PoolManager
+
+rest_client = base.RESTClient()
+
+def get_config(group, option):
+    CONF = cfg.CONF
+    return CONF.group.option
+
+
+def get_host():
+    return RoundRobinScheduler().get_host()        
+
+
+def get_link(url, protocol):
+    host = get_host()
+    return link.make_url(host=host, protocol=protocol, url=url)
+
+    
+def dispatch_get_request(url, protocol, query_params=None):
+    ln = get_link(url, protocol)
+
+    if query_params:
+        query=link.make_query_url(query_params)
+        ln = ln + '?' + query
+
+    return rest_client.GET(ln)
```

Comment:
check whether we can pass query here rather than making query url here we can pass  to the GET method if possible.